ci: Python unit-test suite is not a required check — red tests merged to Dev_new_gui twice

## The Python unit-test suite is not a required status check

Found while working #12647: two of its own merged slices (#13126, #13130) left
**five tests failing on `Dev_new_gui` itself**, and nothing blocked either merge.

Reproduced on the base branch (not on a feature branch):

```
$ PYTHONPATH=$PWD python3 -m pytest autobot_shared/user_management/models/base_test.py -q
FAILED autobot_shared/user_management/models/base_test.py::test_shim_exists[backend]
FAILED autobot_shared/user_management/models/base_test.py::test_shim_exists[slm]
FAILED autobot_shared/user_management/models/base_test.py::test_shim_resolves_to_the_shared_classes[backend]
FAILED autobot_shared/user_management/models/base_test.py::test_shim_resolves_to_the_shared_classes[slm]
FAILED autobot_shared/user_management/models/base_test.py::test_both_backends_share_one_implementation
5 failed, 5 passed
```

```
$ PYTHONPATH=$PWD:$PWD/autobot-backend python3 -m pytest \
    autobot-backend/user_management/middleware/rbac_denial_audit_test.py -q
FAILED ...::TestAuditVocabulary::test_permission_denied_action_exists
FAILED ...::TestAuditVocabulary::test_endpoint_resource_type_exists
FAILED ...::TestAuditVocabulary::test_values_match_the_slm_backend
3 failed, 6 passed
```

## Why nothing caught it

The required checks on `Dev_new_gui` are:

```
smoke-test, code-quality, startup-import-smoke, Unit & Integration Tests,
No open blocks-merge issues reference this PR, No commit trailers,
verify-generated-types, migration-matrix, verify-precommit-config, api-wiring
```

`Unit & Integration Tests` reads like the Python suite but is **not** — it is
defined in `.github/workflows/frontend-test.yml` (plus a skip-success shim job
of the same name, #10527). It gates the frontend.

The Python suite runs in `.github/workflows/ci.yml` under the job
`security-tests` (workflow *AutoBot CI/CD Pipeline*), which covers
`autobot-backend autobot_shared autobot-tts-worker repo_tests` and then
`autobot-slm-backend`, with a `--cov-fail-under=70` gate. **That job's context
is not in the required list**, so a PR can merge with the entire Python suite
red. On #13126 and #13130 the reported `Unit & Integration Tests: SUCCESS` was
the frontend job.

## Impact

Any red Python test — backend, SLM backend, or `autobot_shared` — merges
without resistance. The two cases above sat on `Dev_new_gui` for hours; both
were only noticed because a later slice of the same issue ran the files.

## Suggested fix

Add the `security-tests` job's check context (workflow *AutoBot CI/CD
Pipeline*) to the required status checks for `Dev_new_gui`. Worth considering
at the same time:

- The job name `security-tests` is misleading for what is the main Python
  unit-test + coverage gate; renaming it would make the required-check list
  self-describing and avoid the `Unit & Integration Tests` name collision that
  made this gap look covered.
- `ci.yml` sets `python-version: '3.14'` under a step named
  `Set up Python 3.12`; the comment and the value disagree.

## Already fixed here

Both failing test files are repaired in the #12647 PR that found this
(`fix(user-management): repair two test files left red by earlier #12647
slices`). This issue tracks the **gate**, not those two files.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -44,7 +44,7 @@ jobs:
               - 'autobot-frontend/**'
               - '.github/workflows/ci.yml'
 
-  security-tests:
+  python-suite:
     needs: changes
     if: needs.changes.outputs.python == 'true'
     runs-on: ubuntu-latest
@@ -109,41 +109,13 @@ jobs:
           db: 0
         EOF
 
-    - name: Run code quality checks
-      run: |
-        echo "Running code quality checks..."
-
-        # Code formatting check — must match project standard (pyproject.toml
-        # [tool.black] line-length=120). The previous --line-length=88 disagreed
-        # with the codebase and reported 1864 files "would be reformatted"
-        # because long lines that legitimately fit within 120 exceeded the
-        # stricter 88. (#7496)
-        echo "Checking code formatting with black..."
-        black --check autobot-backend/ autobot-slm-backend/ autobot_shared/ --line-length=120
-
-        # Import sorting check — reads pyproject.toml for profile, src_paths, known_first_party (#2679)
-        echo "Checking import sorting with isort..."
-        isort --check-only --settings-path=. autobot-backend/ autobot-slm-backend/ autobot_shared/
-
-        # Linting (uses project .flake8 config — same as pre-commit)
-        echo "Running flake8 linter..."
-        flake8 --config=.flake8 autobot-backend/ autobot-slm-backend/ autobot_shared/
-
-    - name: Run security analysis
-      run: |
-        echo "🔒 Running security analysis..."
-
-        # Security vulnerability scan — fail on medium+ severity/confidence.
-        # -c .bandit applies the centralized skip list (B101/B104/B105/B106/B108/...)
-        # so known false positives don't fail the gate, matching code-quality.yml
-        # and security.yml (#9694). Real medium+ findings still exit non-zero.
-        bandit -c .bandit -r autobot-backend/ autobot-slm-backend/ autobot_shared/ \
-          --severity-level medium --confidence-level medium \
-          -f json -o bandit-report.json
-        if [ -f bandit-report.json ]; then
-          echo "Security report generated"
-          python -m json.tool < bandit-report.json || true
-        fi
+    # Static checks (black / isort / flake8 / bandit) deliberately live in
+    # code-quality.yml, NOT here (#13162). They were duplicated across both
+    # workflows over the same three trees; code-quality.yml's bandit has no
+    # severity floor where this one filtered to medium+, so it is the stricter
+    # of the two and nothing is lost by removing this copy. Keeping them here
+    # also meant four checks that finish in minutes were gated behind a
+    # ~105-minute pytest run in the same job.
 
     - name: Run unit tests with coverage gate
       run: |
@@ -276,10 +248,10 @@ jobs:
 
   deployment-check:
     runs-on: ubuntu-latest
-    needs: [changes, security-tests, frontend-tests]
+    needs: [changes, python-suite, frontend-tests]
     if: |
       (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/Dev_new_gui') &&
-      (needs.security-tests.result == 'success' || needs.security-tests.result == 'skipped') &&
+      (needs.python-suite.result == 'success' || needs.python-suite.result == 'skipped') &&
       (needs.frontend-tests.result == 'success' || needs.frontend-tests.result == 'skipped')
 
     steps:
@@ -392,7 +364,7 @@ jobs:
 
   notify:
     runs-on: ubuntu-latest
-    needs: [changes, security-tests, frontend-tests, deployment-check]
+    needs: [changes, python-suite, frontend-tests, deployment-check]
     if: always()
 
     steps:
@@ -401,7 +373,7 @@ jobs:
         echo "🎯 CI/CD Pipeline Results:"
         echo "=========================="
 
-        if [ "${{ needs.security-tests.result }}" == "success" ]; then
+        if [ "${{ needs.python-suite.result }}" == "success" ]; then
           echo "✅ Security tests: PASSED"
         else
           echo "❌ Security tests: FAILED"
@@ -420,7 +392,7 @@ jobs:
         fi
 
         echo ""
-        if [ "${{ needs.security-tests.result }}" == "success" ] &&
+        if [ "${{ needs.python-suite.result }}" == "success" ] &&
            [ "${{ needs.frontend-tests.result }}" == "success" ] &&
            [ "${{ needs.deployment-check.result }}" == "success" ]; then
           echo "🎉 All checks passed! AutoBot is ready for deployment."

